packages/connector-saml/src/constant.ts

```diff
@@ -28,3 +27,128 @@ export const authnRequestBinding = ['HTTP-Redirect'] as const;
+  },
+  {
+    type: ConnectorConfigFormItemType.MultilineText,
+    label: 'idpMetadataXml',
+    key: 'idpMetadataXml',
+    required: true,
+  },
+  {
+    type: ConnectorConfigFormItemType.Text,
+    label: 'assertionConsumerServiceUrl',
+    key: 'assertionConsumerServiceUrl',
+    required: true,
+  },
+  {
+    type: ConnectorConfigFormItemType.Select,
+    label: 'requestSignatureAlgorithm',
+    key: 'requestSignatureAlgorithm',
+    selectItems: [
+      { value: 'http://www.w3.org/2000/09/xmldsig#rsa-sha1', title: 'RSA SHA1' },
+      {
+        value: 'http://www.w3.org/2001/04/xmldsig-more#rsa-sha256',
+        title: 'RSA SHA256',
+      },
+      {
+        value: 'http://www.w3.org/2001/04/xmldsig-more#rsa-sha512',
+        title: 'RSA SHA512',
+      },
+    ],
+    defaultValue: 'http://www.w3.org/2001/04/xmldsig-more#rsa-sha256',
+  },
+  {
+    type: ConnectorConfigFormItemType.Select,
+    label: 'messageSigningOrder',
+    key: 'messageSigningOrder',
+    selectItems: [
+      { value: 'sign-then-encrypt', title: 'Sign then encrypt' },
+      {
+        value: 'encrypt-then-sign',
+        title: 'Encrypt then sign',
+      },
+    ],
+    defaultValue: 'sign-then-encrypt',
+  },
+  {
+    type: ConnectorConfigFormItemType.Switch,
+    label: 'signAuthnRequest',
+    key: 'signAuthnRequest',
+    defaultValue: false,
+  },
+  {
+    type: ConnectorConfigFormItemType.MultilineText,
+    label: 'privateKey',
+    key: 'privateKey',
+    required: true,
+    showConditions: [{ targetKey: 'signAuthnRequest', expectValue: true }],
+    placeholder:
+      '-----BEGIN RSA PRIVATE KEY-----\n[private-key-content]\n-----END RSA PRIVATE KEY-----',
+  },
+  {
+    type: ConnectorConfigFormItemType.Text,
+    label: 'privateKeyPass',
+    key: 'privateKeyPass',
+    showConditions: [{ targetKey: 'signAuthnRequest', expectValue: true }],
+  },
+  {
+    type: ConnectorConfigFormItemType.Switch,
+    label: 'encryptAssertion',
+    key: 'encryptAssertion',
+    defaultValue: false,
+  },
+  {
+    type: ConnectorConfigFormItemType.MultilineText,
+    label: 'encPrivateKey',
+    key: 'encPrivateKey',
+    required: true,
+    showConditions: [{ targetKey: 'encryptAssertion', expectValue: true }],
+    placeholder:
+      '-----BEGIN RSA PRIVATE KEY-----\n[private-key-content]\n-----END RSA PRIVATE KEY-----',
+  },
+  {
+    type: ConnectorConfigFormItemType.Text,
+    label: 'encPrivateKeyPass',
+    key: 'encPrivateKeyPass',
+    showConditions: [{ targetKey: 'encryptAssertion', expectValue: true }],
+  },
+  {
+    type: ConnectorConfigFormItemType.Text,
+    label: 'nameIDFormat',
+    key: 'nameIDFormat',
+  },
+  {
+    type: ConnectorConfigFormItemType.Number,
+    label: 'timeout',
+    key: 'timeout',
+    defaultValue: 5000,
+  },
+  {
+    type: ConnectorConfigFormItemType.Json,
+    label: 'profileMap',
+    key: 'profileMap',
+    defaultValue: '{}',
+    required: true,
```

Comment:
You can take a look at the definition of `profileMap` in `types.ts`. Just a reminder: if the `profileMap` is invalid, we may drop it when upsert to connector DB.

Reply:
The config guard will handle this.

Reply:
Per offline discussion, will throw an error when the input value does not fit the type definition.